### PR TITLE
fix cache invalidation in progress plugin

### LINKED ISSUE #14946: Using ProgressPlugin with MiniCssExtractPlugin causes FS cache pack to be invalidated
# Bug report

**What is the current behavior?**
It appears that when using `ProgressPlugin`, `MiniCssExtractPlugin` and File System cache, the first cache pack gets invalidated with the following reason `Pack got invalid because of write to: ProgressPlugin|counts` 

**If the current behavior is a bug, please provide the steps to reproduce.**

```js
const { resolve } = require("path");
const { ProgressPlugin } = require("webpack");
const MiniCssExtractPlugin = require("mini-css-extract-plugin");

module.exports = {
  mode: "development",
  entry: {
    styles: ["./style.css"],
  },
  output: {
    path: resolve(__dirname, "dist"),
    filename: "[name].js",
  },
  plugins: [new MiniCssExtractPlugin(), new ProgressPlugin()],
  module: {
    rules: [
      {
        test: /\.css$/i,
        use: [MiniCssExtractPlugin.loader, "css-loader"],
      },
    ],
  },
  infrastructureLogging: {
    level: "verbose",
  },
  cache: {
    profile: true,
    type: "filesystem",
    maxMemoryGenerations: 1,
  },
};

```

```
➜  webpack-test yarn webpack              
$ /Users/cli-reproductions/webpack-test/node_modules/.bin/webpack
    [webpack-cli] Compiler starting... 
    [webpack-cli] Compiler is using config: '/Users/cli-reproductions/webpack-test/webpack.config.js'
<t> [webpack.cache.PackFileCacheStrategy] restore cache container: 35.202082 ms
    [IdleFileCachePlugin] Initial cache was generated and cache will be persisted in 5s.
    [webpack-cli] Compiler finished
    [webpack.cache.PackFileCacheStrategy] Pack got invalid because of write to: ResolverCachePlugin|normal|dependencyType=|esm|path=|/Users/cli-reproductions/webpack-test|request=|./style.css
    [webpack.cache.PackFileCacheStrategy] Storing pack...
<t> [webpack.cache.PackFileCacheStrategy] resolve build dependencies: 762.821528 ms
<t> [webpack.cache.PackFileCacheStrategy] snapshot build dependencies: 21.222197 ms
    [webpack.cache.PackFileCacheStrategy] 27 fresh items in cache put into pack 0
<t> [webpack.cache.PackFileCacheStrategy] store pack: 27.86769 ms
    [webpack.cache.PackFileCacheStrategy] Stored pack (27 items, 1 files, 0 MiB)
asset styles.js 2.03 KiB [compared for emit] (name: styles)
asset styles.css 222 bytes [compared for emit] (name: styles)
Entrypoint styles 2.25 KiB = styles.css 222 bytes styles.js 2.03 KiB
orphan modules 2.76 KiB (javascript) 937 bytes (runtime) [orphan] 7 modules
runtime modules 274 bytes 1 module
cacheable modules 50 bytes (javascript) 21 bytes (css/mini-extract)
  ./style.css 50 bytes [built] [code generated]
  css ./node_modules/css-loader/dist/cjs.js!./style.css 21 bytes [built] [code generated]
webpack 5.64.4 compiled successfully in 281 ms
✨  Done in 1.69s.


➜  webpack-test yarn webpack 
$ /Users/cli-reproductions/webpack-test/node_modules/.bin/webpack
    [webpack-cli] Compiler starting... 
    [webpack-cli] Compiler is using config: '/Users/cli-reproductions/webpack-test/webpack.config.js'
<t> [webpack.cache.PackFileCacheStrategy] restore cache container: 59.647659 ms
<t> [webpack.cache.PackFileCacheStrategy] check build dependencies: 43.416812 ms
<t> [webpack.cache.PackFileCacheStrategy] restore cache content metadata: 2.01511 ms
    [webpack.cache.PackFileCacheStrategy] starting to restore cache content 0 (32.8 KiB) because of request to: ProgressPlugin|counts
<t> [webpack.cache.PackFileCacheStrategy] restore cache content 0 (32.8 KiB): 10.16656 ms
    [IdleFileCachePlugin] Initial cache was generated and cache will be persisted in 5s.
    [webpack-cli] Compiler finished
    [webpack.cache.PackFileCacheStrategy] Pack got invalid because of write to: ProgressPlugin|counts
    [webpack.cache.PackFileCacheStrategy] Storing pack...
    [webpack.cache.PackFileCacheStrategy] 1 fresh items in cache put into pack 1
<t> [webpack.cache.PackFileCacheStrategy] store pack: 48.41304 ms
    [webpack.cache.PackFileCacheStrategy] Stored pack (27 items, 2 files, 0 MiB)
asset styles.js 2.03 KiB [compared for emit] (name: styles)
asset styles.css 222 bytes [compared for emit] (name: styles)
Entrypoint styles 2.25 KiB = styles.css 222 bytes styles.js 2.03 KiB
cached modules 50 bytes (javascript) 21 bytes (css/mini-extract) 274 bytes (runtime) [cached] 3 modules
webpack 5.64.4 compiled successfully in 278 ms
✨  Done in 0.94s.
```

**What is the expected behavior?**
I expect that since the are no file changes the pack is not invalidated. 

**Other relevant information:**
webpack version:  5.65.0
Node.js version: 14.18.1
Operating System: darwin x64
Additional tools:

### PR DESCRIPTION
**Pull request by vankop**
**What kind of change does this PR introduce?**
fixes #14946 

**Did you add tests for your changes?**
yes

**Does this PR introduce a breaking change?**
no

**What needs to be documented once your changes are merged?**
nothing

### CODE REVIEW

File: lib/Compilation.js [hunk 1]
```diff
@@ -1665,12 +1666,22 @@ BREAKING CHANGE: Asset processing hooks in Compilation has been merged into a si
...
+					const entry = {
 						factory: factoryCacheKey2,
 						dependencies: list,
 						context: dep.getContext(),
 						originModule: module
-					});
+					};
+
+					if (
+						"resourceResolveData" in module &&
```
Comment (vankop) [inline lib/Compilation.js:1677]: probably not a good solution...

File: lib/ProgressPlugin.js [hunk 2]
```diff
@@ -348,10 +349,16 @@ class ProgressPlugin {
...
+			invalidatedCache = false;
+		});
+
+		compiler.cache.hooks.store.tap("ProgressPlugin", () => {
+			invalidatedCache = true;
 		});
 
 		compiler.hooks.afterCompile.tapPromise("ProgressPlugin", compilation => {
-			if (compilation.compiler.isChild()) return Promise.resolve();
+			if (compilation.compiler.isChild() || !invalidatedCache)
```
Comment (vankop) [inline lib/ProgressPlugin.js:360]: this will not cache at all.. we can check `oldData` on line 364, but when update this info?

**Comment by alan-agius4** [comment]
Any chance we can get this in!

**Comment by alexander-akait** [comment]
@alan-agius4 Yea, sorry for delay, in near future

**Comment by vankop** [comment]
only 1 solution could be store this in memory locally not in "common" cache system..

**Comment by vankop** [comment]
when using default system, we anyway need to invalidate at some point

**Comment by vankop** [comment]
also one option is to invalidate cache only when e.g. difference in cache will be greater then X

**Pull request closed without merging**